netsim/ansible/templates/bgp/srlinux.macro.j2

```diff
@@ -25,28 +25,36 @@
       action:
         policy-result: next-policy
 {%   endif %}
-{%   if 'ibgp' in type and nh %}
+{%   if 'ibgp' in type and nhs %}
     - name: rr-next-hop-unchanged
       match:
         protocol: bgp
         bgp:
           community-set: ibgp-mark
       action:
         policy-result: next-policy
+        bgp:
+          communities:
+            remove:
+              ibgp-mark
     - name: ebgp-next-hop-self
       match:
         protocol: bgp
       action:
         policy-result: next-policy
         bgp:
           next-hop:
-            set: "{{ nh }}"
+            set: self
```

Comment:
@ipspace I am curious, what is the use case here where the nhs is needed on the ebgp session? I believe 4271 rules should be applied by default on eBGP sessions.

Or is it some redistribution use case?

Reply:
It's not on an EBGP session. The policy is used on IBGP sessions, and we cannot use the per-neighbor/AF setting because that would also impact the reflected IBGP routes.

The statement is called "ebgp-next-hop-self" because we have to implement the next hop change on EBGP routes. Any better ideas to rename the statement?

Reply:
purely up to you, maybe I'd name it `bgp-next-hop-self`, as the policy is universally matching on the bgp protocol

Reply:
I seem to recall having tested this on iBGP EVPN sessions, and I believe I saw that next hop self wasn't being applied even if the policy asked for it. Could be different in recent releases, but we may want to test for that

Reply:
If you're saying "I don't know whether the action.bgp.next-hop.set: self would work", it does (I'm so glad I wasted so much time writing integration tests), or were you thinking about applying next-hop-self to the peer group? That doesn't work either, the next hop is still changed on reflected routes.